cdap-ui/app/cdap/components/DataPrep/helper.js

```diff
@@ -45,19 +45,21 @@ export function isCustomOption(selectedOption) {
   return selectedOption.substr(0, 6) === 'CUSTOM';
 }
 
-export function setPopoverOffset(element, footerHeight = 54) {
+export function setPopoverOffset(element, popoverLevel = 'second-level-popover') {
   let elem = element;
   let elemBounding = elem.getBoundingClientRect();
-  const FOOTER_HEIGHT = footerHeight;
 
-  let popover = document.getElementsByClassName('second-level-popover');
+  let popover = document.getElementsByClassName(popoverLevel);
   let popoverHeight = popover[0].getBoundingClientRect().height;
   let tableContainerScroll = document.getElementById('dataprep-table-id').scrollTop;
   let popoverMenuItemTop = elemBounding.top;
-  let bodyBottom = document.body.getBoundingClientRect().bottom - FOOTER_HEIGHT;
```

Comment:
I believe there was a problem with the calculation if you remove this footer height if the second level popover is after the scrollable popover item on the main popover. ie. smaller screen, and `Define Variable`